src/languageTools/DefaultProviders.js

```diff
@@ -404,8 +405,82 @@ define(function (require, exports, module) {
         return this._results.get(filePath);
     };
 
+    function ReferencesProvider(client) {
+        this.client = client;
+    }
+
+    ReferencesProvider.prototype.hasReferences = function() {
+        if (!this.client) {
+            return false;
+        }
+
+        var serverCapabilities = this.client.getServerCapabilities();
+        if (!serverCapabilities || !serverCapabilities.referencesProvider) {
+            return false;
+        }
+
+        return true;
+    };
+
+    ReferencesProvider.prototype.getReferences = function(hostEditor, curPos) {
+        var editor = hostEditor || EditorManager.getActiveEditor(),
+            pos = curPos || editor ? editor.getCursorPos() : null,
+            docPath = editor.document.file._path,
+            result = $.Deferred();
+
+        if (this.client) {
+            this.client.findReferences({
+                filePath: docPath,
+                cursorPos: pos
+            }).done(function(msgObj){
+                    if(msgObj && msgObj.length) {
+                        var referenceModel = {};
+                        referenceModel.results = {};
+                        referenceModel.numFiles = 0;
+                        var fulfilled = 0;
+                        msgObj.forEach((element, i) => {
+                            var filePath = PathConverters.uriToPath(element.uri);
+                            DocumentManager.getDocumentForPath(filePath)
+                                .done(function(doc) {
+                                    var startRange = {line: element.range.start.line, ch: element.range.start.character};
```

Comment:
Move this transformation logic to another function so this can be used outside of the DefaultProviders, if necessary.